dotCMS/src/main/java/com/dotcms/rest/api/v1/secret/ServiceIntegrationHelper.java

```diff
@@ -0,0 +1,282 @@
+                .getAvailableServiceDescriptors(user);
+        final List<Host> hosts = hostAPI.findAll(user, false);
+        for (final ServiceDescriptor serviceDescriptor : serviceDescriptors) {
+            final String serviceKey = serviceDescriptor.getServiceKey();
+            final long configurationsCount = computeHostsWithConfigurations(serviceKey, hosts, user).size();
+            viewsBuilder.add(new ServiceIntegrationView(serviceDescriptor, configurationsCount));
+        }
+        return viewsBuilder.build();
+    }
+
+    /**
+     * This gets you a view composed of the service-key and all the hosts that have configurations
+     * @param serviceKey unique service id for the given host.
+     * @param user Logged in user
+     * @return view
+     * @throws DotSecurityException
+     * @throws DotDataException
+     */
+    Optional<ServiceIntegrationHostView> getServiceIntegrationHostView(final String serviceKey,
+            final User user)
+            throws DotSecurityException, DotDataException {
+
+        final Optional<ServiceDescriptor> serviceDescriptorOptional = serviceIntegrationAPI
+                .getServiceDescriptor(serviceKey, user);
+        if (serviceDescriptorOptional.isPresent()) {
+            final ImmutableList.Builder<HostView> hostViewBuilder = new ImmutableList.Builder<>();
+
+            final ServiceDescriptor serviceDescriptor = serviceDescriptorOptional.get();
+            final List<Host> allAvailableHosts = hostAPI.findAll(user, false);
+            final List<Host> hostsWithConfigurations = computeHostsWithConfigurations(serviceKey, allAvailableHosts, user);
+            for (final Host host : hostsWithConfigurations) {
+                hostViewBuilder.add(new HostView(host.getIdentifier(), host.getHostname()));
+            }
+            return Optional.of(
+               new ServiceIntegrationHostView(
+                  new ServiceIntegrationView(serviceDescriptor, hostsWithConfigurations.size()),
+                  hostViewBuilder.build()
+               )
+            );
+        }
+        return Optional.empty();
+    }
+
+    /**
+     * This gives you a detailed view with all the configuration secrets for a given service-key host pair.
+     * @param serviceKey unique service id for the given host.
+     * @param hostId Host Id
+     * @param user Logged in user
+     * @return view
+     * @throws DotSecurityException
+     * @throws DotDataException
+     */
+    Optional<ServiceIntegrationDetailedView> getServiceIntegrationHostDetailedView(
+            final String serviceKey, final String hostId,
+            final User user)
+            throws DotSecurityException, DotDataException {
+        final Optional<ServiceDescriptor> serviceDescriptorOptional = serviceIntegrationAPI
+                .getServiceDescriptor(serviceKey, user);
+        if (serviceDescriptorOptional.isPresent()) {
+            final ServiceDescriptor serviceDescriptor = serviceDescriptorOptional.get();
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Avoid declaring a variable if it is unreferenced before a possible exit point.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4623512)